src/FileFormats.Minidump/MinidumpVirtualAddressSpace.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace FileFormats.Minidump
+{
+    public class MinidumpVirtualAddressSpace : IAddressSpace
+    {
+        private IAddressSpace _addressSpace;
+        private ReadOnlyCollection<MinidumpSegment> _segments;
+        private ulong _length;
+
+        public ulong Length
+        {
+            get
+            {
+                return _length;
+            }
+        }
+
+        public MinidumpVirtualAddressSpace(ReadOnlyCollection<MinidumpSegment> segments, IAddressSpace addressSpace)
+        {
+            _addressSpace = addressSpace;
+            _segments = segments;
+            MinidumpSegment last = segments.Last();
+            _length = last.VirtualAddress + last.Size;
+        }
+
+        public uint Read(ulong position, byte[] buffer, uint bufferOffset, uint count)
+        {
+            if (count == 0)
+                return 0;
+            
+            MinidumpSegment seg = FindSegment(position);
+            if (seg == null)
+                return 0;
+            
+            // TODO: What if they read past the end of the segment?
```

Comment:
I suppose, but I worry about the number of IAddressSpace instances I would need to create...  Similarly, you didn't use piecewise address space in Elf either...ELFVirtualAddressSpace.

I also worry your implementation (both in piecewise and elf) does a linear search for every read...that's bound to get slow, but something that's fixable.